src/CoreWCF.ConfigurationManager/src/CoreWCF/Configuration/AddressHeaderCollectionElement.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Configuration;
+using System.Security;
+using System.Xml;
+using CoreWCF.Channels;
+
+namespace CoreWCF.Configuration
+{
+    public sealed class AddressHeaderCollectionElement : ServiceModelConfigurationElement
+    {
+        public AddressHeaderCollectionElement()
+        {
+        }
+
+        internal void Copy(AddressHeaderCollectionElement source)
+        {
+            if (source == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("source");
+            }
+
+            PropertyInformationCollection properties = source.ElementInformation.Properties;
+            if (properties[ConfigurationStrings.Headers].ValueOrigin != PropertyValueOrigin.Default)
+            {
+                this.Headers = source.Headers;
+            }
+        }
+
+        [ConfigurationProperty(ConfigurationStrings.Headers, DefaultValue = null)]
+        public AddressHeaderCollection Headers
+        {
+            get
+            {
+                AddressHeaderCollection retVal = (AddressHeaderCollection)base[ConfigurationStrings.Headers];
+                if (null == retVal)
+                {
+                    retVal = new AddressHeaderCollection();
+                }
+                return retVal;
+            }
+            set
+            {
+                if (value == null)
+                {
+                    value = new AddressHeaderCollection();
+                }
+                base[ConfigurationStrings.Headers] = value;
+            }
+        }
+
+        [SecuritySafeCritical]
+        protected override void DeserializeElement(XmlReader reader, bool serializeCollectionKey)
```

Comment:
`SecuritySafeCritical` was needed for an old now deprecated .NET Framework feature called Code Access Security. The attribute only exists in .NET Core for compatibility and doesn't do anything. Remove all usages of this attribute.